src/index.js

```diff
@@ -96,13 +94,35 @@ VERBS.concat('any').forEach(function(method) {
   };
 });
 
+function findInHandlers(method, handlers, handler) {
+  var index = -1;
+  for (var i = 0; i < handlers[method].length; i += 1) {
+    var item = handlers[method][i];
+    var isReplyOnce = item.length === 7;
+    var isSame = (
+      item[0] === handler[0] &&
+      item[1] === handler[1] &&
```

Comment:
Parameters and headers need to be checked with deep equality rather than only checking the reference (this library already uses `deep-equal`, check utils.js).
You can verify if it works correctly with this test case:

```javascript
  it('allows overwriting mocks with parameters', function() {
    mock
      .onGet('/users', { params: { searchText: 'John' } })
      .reply(500)
      .onGet('/users', { params: { searchText: 'John' } })
      .reply(200, { id: 1 });

    return instance.get('/users', { params: { searchText: 'John' } }).then(function(response) {
      expect(repsonse.code).to.equal(200);
    });
  });
```
A similar test case is needed for headers as well.

Reply:
Added deep equality check and test for params and headers